spring-rabbit/src/main/java/org/springframework/amqp/rabbit/support/CorrelationData.java

```diff
@@ -28,18 +28,23 @@
  */
 public class CorrelationData {
 
-	private String id;
+	private volatile String id;
 
 	public CorrelationData(String id) {
 		this.id = id;
 	}
 
 	public String getId() {
-		return id;
+		return this.id;
+	}
+
+	protected void setId(String id) {
```

Comment:
Oh! No! 
You have just introduced this setter :innocent: 
Why don't make it just `public` and avoid that `MutableCorrelationData` and just have the `if (this.correlationData != null) {` in the `CorrelationMessagePostProcessor`?

Although that may be just my tired mind to be so nit-pick...